lib/portage/package/ebuild/config.py

```diff
@@ -2204,7 +2204,7 @@ def setcpv(self, mycpv, use_cache=None, mydb=None):
                 # "test" is in IUSE and USE=test is masked, so execution
                 # of src_test() probably is not reliable. Therefore,
                 # temporarily disable FEATURES=test just for this package.
-                self["FEATURES"] = " ".join(x for x in self.features if x != "test")
+                self.features.remove("test")
```

Comment:
This code is workaround from a time when ebuilds were not setting `RESTRICT="!test? ( test )` consistently. I think we could simply drop this hack at this point.

Reply:
that doesn't really make sense, nor is it PMS compliant

PMS states that `RESTRICT` can only be a [simple string](https://projects.gentoo.org/pms/8/pms.html#x1-730008.2) with [specific tokens](https://projects.gentoo.org/pms/8/pms.html#x1-680007.3.6).  anything else is a PM extension and doesn't belong in the Gentoo overlay.  specifically, writing `flag? ( ... )` is only permitted in `specification style variables` which `RESTRICT` is not.

PMS also does not guarantee that `USE=test` is set when running tests -- it doesn't mention that flag anywhere that i can see.  that's a portage-specific convention.  any PM that wants to implement `src_test` support in a reasonable/automatic way for the user has to do the same, but it'd also be perfectly acceptable for the PM to require the user explicitly set `USE=test` and be compliant.

should PMS do better ?  sure.  but since we're squarely in the land of "portage-defined behavior", portage should be doing the heavy lifting here.  otherwise you're suggesting every single ebuild that has `IUSE=test` also has to write `RESTRICT="!test? ( test )"`.  no one is writing ebuilds, let alone verifying them, with the idea that `src_test` will be executed when `USE=-test`.

Raul's change is maintaining status quo.  if you want to change the status quo, do it holistically, and in a separate change.

Reply:
> otherwise you're suggesting every single ebuild that has IUSE=test also has to write RESTRICT="!test? ( test )".

This is already the case. See the Gentoo [policy](https://projects.gentoo.org/qa/policy-guide/other-metadata.html#pg0703).

Reply:
I created https://github.com/gentoo/portage/pull/1116 to remove this logic.